include/pybind11/pytypes.h

```diff
@@ -47,54 +47,65 @@ using tuple_accessor = accessor<accessor_policies::tuple_item>;
 class pyobject_tag { };
 template <typename T> using is_pyobject = std::is_base_of<pyobject_tag, remove_reference_t<T>>;
 
+// clang-format off
 /** \rst
     A mixin class which adds common functions to `handle`, `object` and various accessors.
     The only requirement for `Derived` is to implement ``PyObject *Derived::ptr() const``.
 \endrst */
+// clang-format on
 template <typename Derived>
 class object_api : public pyobject_tag {
     const Derived &derived() const { return static_cast<const Derived &>(*this); }
 
 public:
+    // clang-format off
     /** \rst
         Return an iterator equivalent to calling ``iter()`` in Python. The object
         must be a collection which supports the iteration protocol.
     \endrst */
+    // clang-format on
     iterator begin() const;
     /// Return a sentinel which ends iteration.
     iterator end() const;
 
+    // clang-format off
     /** \rst
         Return an internal functor to invoke the object's sequence protocol. Casting
         the returned ``detail::item_accessor`` instance to a `handle` or `object`
         subclass causes a corresponding call to ``__getitem__``. Assigning a `handle`
         or `object` subclass causes a call to ``__setitem__``.
     \endrst */
+    // clang-format on
     item_accessor operator[](handle key) const;
     /// See above (the only difference is that they key is provided as a string literal)
     item_accessor operator[](const char *key) const;
 
+    // clang-format off
```

Comment:
Why is this necessary?

Reply:
To protect the `\rst` to `\endrst` block enclosed by the `// clang-format off`, `on`.

I added the clang-format meta-comments systematically rather than trying to figure out case by case if they are needed or not. Possibly some could be omitted, but I believe for blocks that are deliberately formatted as rst it's better to systematically exclude them from clang-format.

Note that the sphinx error messages are completely useless if something goes wrong. I spent quite a bit of time figuring out why the Documentation GitHub action was failing. Now I know, but letting clang-format loose on rst is likely to turn into time-consuming surprises for others.